contactform.php

```diff
@@ -434,8 +502,29 @@ public function sendMessage()
         $clientToken = Tools::getValue('token');
         $serverToken = $this->context->cookie->contactFormToken;
         $clientTokenTTL = $this->context->cookie->contactFormTokenTTL;
-
-        if (!($from = trim(Tools::getValue('from'))) || !Validate::isEmail($from)) {
+        $useReCaptcha = (int)Configuration::get(self::USE_RECAPTCHA);
+        $reCaptchaSuccess = false;
+        
+        //If user is not logged in and we want to use reCaptcha
+        if (!$this->context->customer->isLogged() && $useReCaptcha) {
+            //start checking
+            $secretKey = Configuration::get(self::RECAPTCHA_SECRET_KEY);
+            //google recaptch check
+            $response_google=file_get_contents(
+                "https://www.google.com/recaptcha/api/siteverify?secret=".$secretKey.
```

Comment:
```suggestion
                "https://www.google.com/recaptcha/api/siteverify?secret=" . Configuration::get(self::RECAPTCHA_SECRET_KEY).
```

Reply:
I think, that the code will be less readable.
```
$response_google=file_get_contents(
                "https://www.google.com/recaptcha/api/siteverify?secret=".
                    Configuration::get(self::RECAPTCHA_SECRET_KEY).
                "&response=".$_POST['g-recaptcha-response'].
                "&remoteip=".$_SERVER['REMOTE_ADDR']);
```

Besides, Tools::getRemoteAddr() should be used as well as Tools::getValue() instead of $_POST.

I need to test it.